### DISCUSSION TITLE
Door "not responding" after a few seconds

### DISCUSSION THREAD

**Discussion by TomVanHove** [HomeSpan/HomeSpan#695] [Q&A]
Hi all!

I'm working with an ESP32 dev kit with a servo SG90 to control a door. 

When the door is opening/closing, Apple Homekit shows "no response" in the app after the motor is running several seconds.

If I close Apple Homekit and open it again, the door is effectively open ... (as it should be).

Attached, a video of the 'problem'.

Is there any way to update/refresh Apple Homekit?

Thank you in advance!!

https://github.com/HomeSpan/HomeSpan/assets/43343745/828d2577-3557-4c05-aaf7-d51d979a9dcd

**Comment by HomeSpan** [comment]
How are you operating the door.  Is the Servo running in a background process (what library are you using)?  If not, then homeSpan.poll() won't be able to run while the Servo is running, and this is what causes the temporary no-response.  There are a few ways to address this.  If you are running a Servo library that interferes with, and blocks, the main `loop()` from executing `homeSpan.poll()`, you can try instead to run `homeSpan.poll()` in the background by using `homeSpan.autoPoll()` instead.  See the API for details on how to do this.

Alternatively, HomeSpan has built-in Servo functions that are designed to run in the background without interfering with the communication between HomeSpan and HomeKit.  See the Servo documentation for details.

**Reply by TomVanHove** [reply]
Hi! 

Thanks for your fast response! :)

YES! The main loop is interrupted by the Servo function. The Servo keeps going until an end switch is reached.

I will check the HomeSpan Servo function!

**Reply by TomVanHove** [reply]
It helped removing the loop that interrupts the HomeSpan poll() function.

Thanks a lot!!
